client/src/pages/admin-pages/addFlight/index.js

```diff
@@ -9,90 +9,82 @@ import {
   Button,
   Header,
   SideNav
-} from "../../components";
+} from "../../../components";
 
 import "./addflight.css";
 
 class AddFlight extends Component {
   state = {
     form: {},
-    selectedCity: {},
+    selectedValue: {},
     flightDuration: "",
     succesfullyAdded: false,
-    alert: null
+    alert: false,
+    error: false
   };
 
   handleAddFlight = e => {
     e.preventDefault();
-    const { dateValue, airCraftTypeValue } = this.state.form;
-    const { originValue, destinationValue } = this.state.selectedCity;
+
+    const {
+      flightDuration,
+      selectedValue: { originValue, destinationValue, airCraftTypeValue },
+      form: { dateValue, flightNumberValue }
+    } = this.state;
 
     const data = JSON.stringify({
+      flightDuration,
       originValue,
       destinationValue,
       dateValue,
-      airCraftTypeValue
+      airCraftTypeValue,
+      flightNumberValue
     });
-    fetch("/addFlight_post", {
+    fetch("/api/v1/flight", {
       credentials: "same-origin",
       headers: {
         "content-type": "application/json"
       },
       method: "POST",
       body: data
     })
-      .then(response => response.json(data))
+      .then(response => response.json())
       .then(data => {
         if (data.succesfullyAdded) {
           this.setState({
-            succesfullyAdded: true
+            succesfullyAdded: true,
+            alert: true
```

Comment:
I wonder if this should also set error to false.  That might handle the scenario where you try to submit, it fails for some reason (and you see an error), so you resubmit and it works.  You'd want to show the success message *and* turn off the error message.

Reply:
Nevermind that comment - the user would have already dismissed the error message themselves before clicking submit again.